src/nvc.rs

```diff
@@ -2,6 +2,100 @@ use instruction::*;
 use interconnect::*;
 
 use std::collections::HashSet;
+use std::fmt;
+
+#[derive(Copy, Clone)]
+pub struct NvcCacheEntry {
+    pub tag: u32,
+    pub base_addr: u32,
+    pub subblock_valid: [bool; 2],
+}
+
+impl NvcCacheEntry {
+    fn new() -> NvcCacheEntry {
+        NvcCacheEntry {
+          tag: 0,
+          base_addr: 0,
+          subblock_valid: [false; 2],
+        }
+    }
+}
+
+impl fmt::Display for NvcCacheEntry {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "base_addr: 0x{:08x}, tag: 0x{:08x}, sb1 valid: {:5}, sb2 valid: {:5}", self.base_addr, self.tag, self.subblock_valid[0], self.subblock_valid[1])
+    }
+}
+
+pub struct NvcCache {
+    hits: usize,
+    misses: usize,
+    cache_entries: [NvcCacheEntry; 128],
```

Comment:
nit: I think this should be `Box`'d (as in `Box<[NvcCacheEntry::new(); 128]>`) since this collection of structures could be fairly large, and we don't want to initialize that on the stack. Should only be a kB or so, but that's beyond my typical threshold for this kind of thing :)

Side note: it would still be initialized on the stack and then moved to the heap with `Box::new`, but there's an experimental syntax (just called `box` that's not stable but in the works) which would initialize it on the heap directly, so eventually we could do that properly. But for now, just `Box`'ing this array would be nice I think.